server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1345,7 +1345,7 @@ && shouldSkipReplicationTimer(entry.getKey()) == false
                                 && isShardOnRemoteEnabledNode.apply(routingTable.getByAllocationId(entry.getKey()).currentNodeId())))
                 )
                 .map(entry -> buildShardStats(entry.getKey(), entry.getValue()))
-                .collect(Collectors.toUnmodifiableSet());
+                .collect(Collectors.toSet());
```

Comment:
lets pls create a new collection where required and not make this mutable

Reply:
Sure